Doc/c-api/float.rst

```diff
@@ -45,14 +45,14 @@ Floating Point Objects
 .. c:function:: double PyFloat_AsDouble(PyObject *pyfloat)
 
    Return a C :c:expr:`double` representation of the contents of *pyfloat*.  If
-   *pyfloat* is not a Python floating point object but has a :meth:`__float__`
+   *pyfloat* is not a Python floating point object but has a :meth:`~object.__float__`
    method, this method will first be called to convert *pyfloat* into a float.
-   If ``__float__()`` is not defined then it falls back to :meth:`__index__`.
+   If :meth:`~object.__float__` is not defined then it falls back to :meth:`~object.__index__`.
```

Comment:
Same as above: it's probably not necessary to have a link here, since we link to it in the previous sentence

```suggestion
   If :meth:`!__float__` is not defined then it falls back to :meth:`~object.__index__`.
```